### PR TITLE
Muon shower fix 

### PR DESCRIPTION
**Pull request by jhgoh**
This PR re-applies #12090 which fixes issues in the muon shower variable calculation.
Some unnecessary rearranging of vectors are removed with this fix and eventually it also solves potential problem of random behavior in events with very large muon hit multiplicity.

### CODE REVIEW

**Comment by slava77** [comment]
I was expecting that this comes with a fix to the AbsLessDTheta implementation.

**Comment by davidlange6** [comment]
please test

**Comment by slava77** [comment]
@jhgoh 
please clarify your plans about AbsLessDTheta method

File: RecoMuon/MuonIdentification/src/MuonShowerInformationFiller.cc [hunk 1]
```diff
@@ -838,28 +809,83 @@ void MuonShowerInformationFiller::fillHitsByStation(const reco::Muon& muon) {
...
+
+    // Cluster seeds by global position phi. Best cluster is chosen to give greatest dphi
+    // Sort by phi (complexity = NLogN with enough memory, or = NLog^2N for insufficient mem)
+    stable_sort(muonRecHits[stat].begin(), muonRecHits[stat].end(), LessPhi());
+
+    // Search for gaps (complexity = N)
+    std::vector<size_t> clUppers;
+    for ( size_t ihit = 0; ihit < nhit; ++ihit ) {
+      const size_t jhit = (ihit+1)%nhit;
+      const double phi1 = muonRecHits[stat].at(ihit)->globalPosition().phi();
```
Comment (slava77) [inline RecoMuon/MuonIdentification/src/MuonShowerInformationFiller.cc]: use barePhi, as it is now in the updated baseline code

Comment (slava77) [inline RecoMuon/MuonIdentification/src/MuonShowerInformationFiller.cc]: .. implies to do it everywhere in the file: 's/globalPosition().phi()/globalPosition().barePhi()/'

Comment (jhgoh) [inline RecoMuon/MuonIdentification/src/MuonShowerInformationFiller.cc]: Thank you for comment, I'll replace them.

**Comment by slava77** [comment]
@jhgoh 
did you forget to commit changes to .cc file as well (the .barePhi)

**Comment by jhgoh** [comment]
Updated missing barePhi() changes in .cc file.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/13116	ac044b6
- code changes are in line with the description and have essentially the same update as was reviewed in #12090 (it was not integrated in 74X due to result consistency requirements).
- jenkins tests pass and comparisons with baseline show differences only in muonShowerInformation monitorables. Changes are noticeable, but are consistent with the baseline, as expected: e.g. workflow 1330 from jenkins comparisons
wf1330_mushower_tkmuons_st2_clusize
- originally, modifications to this code were prompted by long event processing issues reported in https://hypernews.cern.ch/HyperNews/CMS/get/recoDevelopment/1392.html. E.g. in "Run 258629, Event 1242504, LumiSection 41" of MET PD the muonShowerInformation was taking a lot of time. Checking in 800pre5, the same event still takes a long time (test job was running for more than 30 min); after the update in this PR the event is processed by muonShowerInformation in about 4 seconds.

**Comment by davidlange6** [comment]
+1

**Pull request merged**